orttraining/orttraining/python/training/ortmodule/_experimental/_load_config_from_json.py

```diff
@@ -0,0 +1,146 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+# _load_config_from_json.py
+
+import json
+import os
+from types import SimpleNamespace
+
+from onnxruntime.capi import _pybind_state as C
+from functools import reduce
+from .._graph_execution_manager import _SkipCheck
+from ..debug_options import DebugOptions, LogLevel
+
+def load_data_from_json(path):
+    """Loads data from the json file path provided."""
+
+    data = None
+    with open(path) as f:
+        data = json.load(f, object_hook=lambda d: SimpleNamespace(**d))
+
+    if data is None:
+        raise RuntimeError(f"No data found in provided json file {path}.")
+
+    return data
+
+def load_propagate_cast_ops(ortmodule_config_accessor, data):
+    """Load PropagateCastOps from json file onto ORTModule"""
+
+    propagate_cast_ops_str = 'PropagateCastOps'
```

Comment:
Not sure whether this was discussed on the design (is there a new version for internal only?), but it seems tricky to use case-sensitive strings with CamelCase. 

Users can make a typo hard to spot. Is there any way users would be warned about 1) incorrect names and 2) unused settings in the json (probably because of a typo)?

Reply:
I can add a check thta runs through the top level json and tries to find any setting that is not understood by the loading functions.

Reply:
cerberus would solve this and all other checks and default values in a single line, but not sure whether we want this dependency. It caused some controversy on ORTTrainer

Reply:
Instead of checking for keys in the json data, I iterate over the json data at all levels. This way, if there is any unknown key, it will result in a ```KeyError```.